pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/DataDictionaryExporter.java

```diff
@@ -0,0 +1,293 @@
+            if (!moduleName.equals(previousModuleName)) {
+                addModuleHeaderRows(moduleName, filterConfig.getParent());
+            }
+            String questionId = null;
+            String parentQuestionId = null;
+            if (filterConfig.getQuestionDef() != null) {
+                if (parentConfig != null) {
+                    parentQuestionId = (String) parentConfig.getQuestionDef().get(ESObjectConstants.STABLE_ID);
+                }
+                questionId = (String) filterConfig.getQuestionDef().get(ESObjectConstants.STABLE_ID);
+            }
+
+            if (!StringUtils.equals(previousParentQuestionId, parentQuestionId) && parentQuestionId != null) {
+                addCompositeQuestionHeaderRow(filterConfig, activityRepeatNum, questionRepeatNum, option, detailName, parentConfig);
+            }
+            if (!StringUtils.equals(questionId, previousQuestionId) && filterConfig.isSplitOptionsIntoColumns()) {
+                addSplitOptionsHeaderRow(filterConfig, activityRepeatNum, questionRepeatNum, option, detailName, parentConfig);
+            }
+
+            if (detailName != null) {
+                addAdditionalDetailRow(filterConfig, activityRepeatNum, questionRepeatNum, option, detailName, parentConfig);
+            } else if (option != null) {
+                addSplitOptionsRow(filterConfig, activityRepeatNum, questionRepeatNum, option, null, parentConfig);
+            } else {
+                addRegularVariableRow(filterConfig, activityRepeatNum, questionRepeatNum, null, null, parentConfig);
+            }
+            previousModuleName = moduleName;
+            previousQuestionId = questionId;
+            previousParentQuestionId = parentQuestionId;
+        }
+    }
+
+    protected void addModuleHeaderRows(String moduleName, ModuleExportConfig moduleExportConfig) {
+        // two blank rows
+        addRowToSheet();
+        addRowToSheet();
+
+        SXSSFRow moduleNameRow = addRowToSheet();
+        moduleNameRow.setRowStyle(boldStyle);
+        moduleNameRow.createCell(0).setCellValue(moduleName.toUpperCase());
+
+        if (moduleExportConfig.getNumMaxRepeats() > 1) {
+            SXSSFRow moduleRepeatRow = addRowToSheet();
+            String repeatString = "Up to " + moduleExportConfig.getNumMaxRepeats() +
+                    " entries for this module exist for each participant.\n";
+            repeatString += "Entries are indicated in reverse chronological order.\n";
+            repeatString += "The most recent entry has no suffix, the next-most-recent is suffixed with _2," +
+                    " the next-most-recent is suffixed with _3, etc...\n ";
+            repeatString += "e.g. " + moduleName + ".[QUESTION] is the most recent completion, and " + moduleName +
+                    "_2.QUESTION is the next-most recent completion.";
+            moduleNameRow.createCell(1).setCellValue(repeatString);
```

Comment:
I think having body of this if statement in separated method with meaningful name would be beneficial for quick understanding in future.